adam/language_specific/english/english_language_generator.py

```diff
@@ -96,8 +100,25 @@ class _Generation:
         ] = dict()
 
         def generate(self) -> ImmutableSet[LinearizedDependencyTree]:
-            for _object in self.situation.objects:
-                self._translate_object_to_noun(_object)
+            # For now, only apply quantifiers to object-only situations
+            if self.situation.actions == ImmutableSet.empty():
+                # Get number of objects of each type
+                node_counts: Dict[OntologyNode, int] = dict()
+                for _object in self.situation.objects:
+                    try:
+                        node_counts[_object.ontology_node] += 1
+                    except KeyError:
+                        node_counts.update({_object.ontology_node: 1})
+                object_counts: Dict[SituationObject, int] = {
```

Comment:
I think you can skip `object_counts` and just look below at the count of the node's `ontology_node`

Reply:
Done